libraries/AP_OpticalFlow/AP_OpticalFlow_HereFlow.cpp

```diff
@@ -0,0 +1,102 @@
+#include <AP_HAL/AP_HAL.h>
+
+#if HAL_WITH_UAVCAN
+
+#include <AP_Common/Semaphore.h>
+
+#include "AP_OpticalFlow_HereFlow.h"
+
+#include <AP_BoardConfig/AP_BoardConfig_CAN.h>
+#include <AP_UAVCAN/AP_UAVCAN.h>
+
+#include <com/hex/equipment/flow/Measurement.hpp>
+
+extern const AP_HAL::HAL& hal;
+
+#define debug_flow_uavcan(level_debug, can_driver, fmt, args...) do { if ((level_debug) <= AP::can().get_debug_level_driver(can_driver)) { hal.console->printf(fmt, ##args); }} while (0)
+
+//UAVCAN Frontend Registry Binder
+UC_REGISTRY_BINDER(MeasurementCb, com::hex::equipment::flow::Measurement);
+
+uint8_t AP_OpticalFlow_HereFlow::_node_id = 0;
+AP_OpticalFlow_HereFlow* AP_OpticalFlow_HereFlow::_driver = nullptr;
+AP_UAVCAN* AP_OpticalFlow_HereFlow::_ap_uavcan = nullptr;
+/*
+  constructor - registers instance at top Flow driver
+ */
+AP_OpticalFlow_HereFlow::AP_OpticalFlow_HereFlow(OpticalFlow &flow) :
+    OpticalFlow_backend(flow)
+{
+    if (_driver) {
+        AP_HAL::panic("Only one instance of Flow supported!");
+    }
+    _driver = this;
+}
+
+void AP_OpticalFlow_HereFlow::subscribe_msgs(AP_UAVCAN* ap_uavcan)
+{
+    if (ap_uavcan == nullptr) {
+        return;
+    }
+
+    auto* node = ap_uavcan->get_node();
+
+    uavcan::Subscriber<com::hex::equipment::flow::Measurement, MeasurementCb> *measurement_listener;
+    measurement_listener = new uavcan::Subscriber<com::hex::equipment::flow::Measurement, MeasurementCb>(*node);
+    // Msg Handler
+    const int measurement_listener_res = measurement_listener->start(MeasurementCb(ap_uavcan, &handle_measurement));
+    if (measurement_listener_res < 0) {
+        AP_HAL::panic("UAVCAN Flow subscriber start problem\n\r");
+        return;
+    }
+}
+
+void AP_OpticalFlow_HereFlow::handle_measurement(AP_UAVCAN* ap_uavcan, uint8_t node_id, const MeasurementCb &cb)
+{
+    if (_driver == nullptr) {
+        _ap_uavcan = ap_uavcan;
+        _node_id = node_id;
```

Comment:
Similarly this looks to be write only.

Reply:
Fixed in https://github.com/ArduPilot/ardupilot/pull/11017/commits/6970978f1237353130863dd087089047a0a5c93a